api/helpers.go

```diff
@@ -53,19 +53,28 @@ func getEntryFromPath(ctx context.Context, path string) (plugin.Entry, *errorRes
 		panic("path should never be empty")
 	}
 
+	// Get the registry from context (added by registry middleware).
+	registry := ctx.Value(pluginRegistryKey).(*plugin.Registry)
+	if path == "/" {
+		// Return the registry
+		return registry, nil
+	}
+
 	// Don't interpret trailing slash as a new segment, and ignore optional leading slash
 	path = strings.Trim(path, "/")
 	// Split into plugin name and an optional list of segments.
 	segments := strings.Split(path, "/")
 	pluginName := segments[0]
 	segments = segments[1:]
 
-	// Get the registry from context (added by registry middleware).
-	registry := ctx.Value(pluginRegistryKey).(*plugin.Registry)
 	root, ok := registry.Plugins()[pluginName]
 	if !ok {
 		return nil, pluginDoesNotExistResponse(pluginName)
```

Comment:
Hmm, how important is it to differentiate a non-existent plugin from a non-existent entry? If it's not that important, we could probably move this code over to `findEntry`.

Reply:
I think it's useful to leave it differentiated.